azurerm/resource_arm_kubernetes_cluster_test.go

```diff
@@ -1744,3 +1767,32 @@ resource "azurerm_kubernetes_cluster" "test" {
 }
 `, rInt, location, rInt, rInt, clientId, clientSecret)
 }
+
+func testAccAzureRMKubernetesCluster_nodeResourceGroup(rInt int, clientId string, clientSecret string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  dns_prefix          = "acctestaks%d"
+	node_resource_group = "ExampleResourceGroup"
```

Comment:
to protect against this running in parallel could we make this:

```suggestion
  node_resource_group = "acctestRGAKS-%d"
```

(and then thread through another rInt below, to ensure that this resource group is unique?)